.github/workflows/ios.yml

```diff
@@ -1,173 +1,224 @@
+        - name: Pre-checks - Env is Dev
+          run: |
+            echo "ENV=development" >> $GITHUB_ENV
+            echo "SECRET_NAME=DEV_ENV" >> $GITHUB_ENV
+            echo "APPCENTER_NAME=GoodDollar/GoodDollar-iOS-development" >> $GITHUB_ENV
+            echo "APPCENTER_TOKEN=${{ secrets.APPCENTER_IOS_DEV }}" >> $GITHUB_ENV
+
+        - name: Pre-checks - Env is QA
+          if: ${{ env.TARGET_BRANCH == 'staging' }}
+          run: |
+            echo "ENV=staging" >> $GITHUB_ENV
+            echo "SECRET_NAME=STAGING_ENV" >> $GITHUB_ENV
+            echo "APPCENTER_NAME=GoodDollar/GoodDollar-iOS-staging" >> $GITHUB_ENV
+            echo "APPCENTER_TOKEN=${{ secrets.APPCENTER_IOS_STAGING }}" >> $GITHUB_ENV
+
+        - name: Pre-checks - Env is PROD
+          if: ${{ env.TARGET_BRANCH == 'next' }}
+          run: |
+            echo "ENV=prod" >> $GITHUB_ENV
+            echo "SECRET_NAME=PROD_ENV" >> $GITHUB_ENV
+            echo "APPCENTER_NAME=GoodDollar/GoodDollar-iOS-production" >> $GITHUB_ENV
+            echo "APPCENTER_TOKEN=${{ secrets.APPCENTER_IOS_PROD }}" >> $GITHUB_ENV
+            echo "IS_PROD=true" >> $GITHUB_ENV
+
+        - uses: actions/checkout@v2
+          with:
+            ref: ${{ env.TARGET_BRANCH }}
+        - uses: actions/setup-node@v1
+          with:
+            node-version: 14
+        - name: fix max_user_watches
+          run: |
+            echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
+        - name: add .env secrets
+          env:
+            SENTRYRC: ${{ secrets.sentryrc_file }}
+          run: |
+            env_name="${{ env.ENV }}"
+            echo $env_name
+            cat .env.$env_name
+            echo "adding secrets to .env.$env_name file: ${{ env.SECRET_NAME }}"
+            echo "$SENTRYRC" > android/sentry.properties
+            echo "${{ secrets[env.SECRET_NAME] }}" >> .env.$env_name
+        - uses: actions/checkout@v1
+        - uses: actions/setup-node@v1
+          with:
+            node-version: 10.15.3
+
+        - name: Switch XCode Version
+          run: sudo xcode-select -s /Applications/Xcode_11.2.app
+
+        - name: Setup provisioning profile
+          run: ./.github/secrets/decrypt_secrets.sh
+          env:
+            IOS_CERTIFICATE: ${{ secrets.IOS_CERTIFICATE }}
+            IOS_STORE_PROFILE: ${{ secrets.IOS_STORE_PROFILE }}
+            IOS_ADHOC_PROFILE: ${{ secrets.IOS_ADHOC_PROFILE }}
+            IOS_PROFILE_KEY: ${{ secrets.IOS_PROFILE_KEY }}
+            IS_PROD: ${{ env.IS_PROD }}
+
+        - name: Cache node modules
+          uses: actions/cache@v1
+          with:
+            path: node_modules
+            key: ${{ runner.OS }}-npm-cache-${{ hashFiles('**/package-lock.json') }}
+            restore-keys: |
+              ${{ runner.OS }}-npm-cache-
+        - name: Install npm dependencies
+          run: |
+            npm ci
+        - name: Cache Pods dependencies
```

Comment:
check does exists an dedication github action to install & cache pods
if yes - use it instead